### ISSUE TITLE
include breaks block content / template extending

### ISSUE THREAD

**Issue by fox34** [smarty-php/smarty#215]
Including extending files in an extended file is not possible any more (was working as of Smarty 3.1.13, I do not know the latest working version).

PHP Version: 7.0.4 (seems to be version independent down to 5.4)
Smarty: 3.1.29

Given the following scenario (see attached files for an example):

[include-sample.zip](https://github.com/smarty-php/smarty/files/215350/include-sample.zip)

# Templates:
* templates/start.tpl
* templates/start-extended.tpl (extends start.tpl, as the name states)
* templates/included.tpl
* templates/included-extended.tpl (extends included.tpl, as the name states)

Including "included-extended.tpl" in "start.tpl" leads to wrong block values when displaying start-extended.tpl: start-extended.tpl will be completely ignored.

# Example 1
start.tpl:
```
{include file="included-extended.tpl"}
{block "foo"}This is block "foo" in start.tpl{/block}
```
start-extended.tpl:
```
{extends "start.tpl"}
{block "foo"}This is block "foo" in start-extended.tpl{/block}
```
included.tpl:
```
{block "bar"}This is block "bar" in included.tpl{/block}
```
included-extended.tpl:
```
{extends file="included.tpl"}
{block "bar"}This is block "bar" in included-extended.tpl{/block}
```
Expected output of **start-extended.tpl**:
```
This is block "bar" in included-extended.tpl
This is block "foo" in start-extended.tpl
```

Real output of **start-extended.tpl**:
```
This is block "bar" in included-extended.tpl
This is block "foo" in start.tpl
```

# Example 2
start.tpl:
```
{* include file="included-extended.tpl" *}
{block "foo"}This is block "foo" in start.tpl{/block}
```
The other files have not been changed.

Output **start-extended.tpl** (this is the expected behavior):
```

This is block "foo" in start-extended.tpl
```

# Example 3
Smarty version 3.1.13
Same code as in example 1, this is the expected behavior:
```
This is block "bar" in included-extended.tpl
This is block "foo" in start-extended.tpl
```

**Comment by fox34** [comment]
On first sight, this seems to be related with #200.

**Comment by uwetews** [comment]
Your example does sort break the definition that the {extends} tag shall be first in inheritance templates. See http://www.smarty.net/docs/en/language.function.extends.tpl.
You have it in the include file.
In earlier versions of Smarty resolving template inheritance was a compile time process which had a couple of restrictions. However unintended it did create your expected result.

Since 3.1.29 inheritance is resolved at runtime to get rid of all previous restrictions, but your example is not working any more as start-extended.tpl does see start.tpl as final parent template.

I see your idea behind it and will try to find a solution.

**Comment by uwetews** [comment]
This bug is fixed in the master branch and will later be included in 3.1.30

**Comment by fox34** [comment]
Thanks for your quick reaction and bugfix. Now working as intended. :+1:
